blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -84,10 +84,7 @@ public Integer getWeatherScore() {
 
     @Override
     public BlueRun getLatestRun() {
-        if(job.getLastBuild() == null){
-            return null;
-        }
-        return AbstractRunImpl.getBlueRun(job.getLastBuild(), this);
+        return getRuns().iterator().next();
```

Comment:
This might fail in case, a job is run for the first time and iterator().next() might result in to `NoSuchElementException` 🐛  I think the existing code is correct, where `job.getLastBuild()` returns null if `runs.isEmpty()` returns null value.